membership-attribute-service/app/controllers/AccountController.scala

```diff
@@ -81,8 +81,10 @@ class AccountController(commonActions: CommonActions, override val controllerCom
     def retrieveZuoraSubscription(user: String): Future[ApiError \/ Subscription[P]] = {
       val getSubscriptionData = for {
         sfUser <- EitherT(tp.contactRepo.get(user).map(_.flatMap(_ \/> s"No Salesforce user: $user")))
-        zuoraSubscription <- EitherT(tp.subService.current[P](sfUser).map(subscriptionSelector(subscriptionNameOption, s"Salesforce user $sfUser")))
-      } yield zuoraSubscription
+        validationSub <- EitherT(tp.subService.current[P](sfUser).map(subscriptionSelector(Some(subscriptionName), s"Salesforce user $sfUser")))
+        zuoraCurrentSub <- OptionT(tp.subService.get[P](subscriptionName, isActiveToday = true)).toRight("notFound")
+        _ <- EitherT(Future.successful(if (validationSub.name == zuoraCurrentSub.name) \/-(zuoraCurrentSub) else -\/("Sub does not belong to Salesforce user")))
+      } yield zuoraCurrentSub
```

Comment:
This scalaz/cats code could be difficult to make sense of.  Are there existing tests for this code?  Although I think the difficulty is in parsing the types rather than the logic of it.

Reply:
Attempted to clarify logic in https://github.com/guardian/members-data-api/pull/471/commits/dcad758ac7c83b8719d35c2e129780a9066192b9 Regarding types, not much we can do without major refactoring in membership-common and/or members-data-api to have the types align.